packages/ui/src/views/pipeline-edit/components/graph-implementation/utils/yaml-to-pipeline-graph.tsx

```diff
@@ -16,6 +18,17 @@ import { YamlEntityType } from '../types/yaml-entity-type'
 import { getIconBasedOnStep } from './step-icon-utils'
 import { getNameBasedOnStep } from './step-name-utils'
 
+export const yamlString2Nodes = (
+  yaml: string,
+  options: {
+    selectedPath?: string
+    getStepIcon?: (step: Record<string, any>) => JSX.Element
+  } = {}
+) => {
+  const yamlJson = parse(yaml)
```

Comment:
this should be wrapped in a try-catch? or is that left to the user intentionally?